tests/ui/rfcs/rfc-0000-never_patterns/typeck.rs

```diff
@@ -0,0 +1,118 @@
+#![feature(never_patterns)]
```

Comment:
It might be worth splitting this test file into two: A postive (`// check-pass`) and a negative one.

For one thing, it makes it clear for readers and for another it ensures that the positive test *actually* passes. Due to the existence of `delayed_bug` it's certainly possible that the positive subset of test cases in this file doesn't actually compile if the other errors mask / overshadow delayed bugs.

Reply:
In general I would agree with you, but here it's all match checking stuff which is reliably run independently for each match expression. I find having separate files makes it quite a bit harder to compare what's up, I've been annoyed every time I've encountered files like that in the test suite x)

Reply:
A possible middle ground may be to use revisions: it's still in the same file while being independent

Reply:
@fmease does make a very good and important point: there are more than a few cases where the "positive" part of a test goes untested because of the presence of negative tests that taint an infcx, leak a `ty::Error`, etc.

It takes a lot of trust to hope that "stuff [is] reliably run independently for each match expression", and this is an invariant that can change due to modifications for error suppression, changes to the ordering of queries executed, etc. It's really not worth test cleanliness -- a test's *first and foremost* job is to make sure it's exercising what we expect them to.

I think this should use a revision or split out the tests into a separate file.